### PR TITLE
[FE-2693] Precompile tag validation regex

### PR DESCRIPTION
**Pull request by trevor-fauna**
### Notes
[FE-2693](https://faunadb.atlassian.net/browse/FE-2693)

Addressing out-of-band feedback on https://github.com/fauna/faunadb-js/pull/658

### How to test
`yarn test`

### CODE REVIEW

File: src/_http/index.js [hunk 1]
```diff
@@ -169,6 +169,8 @@ function validateTags(tags) {
   validateTagValues(Object.values(tags))
 }
 
+const AlphaNumPattern = new RegExp(/^\w+$/)
```
Comment (doom-weaver) [inline src/_http/index.js:172]: try putting this at line 6 and i think it should still work mint it just once for any call to HttpClient.

Comment (trevor-fauna) [reply src/_http/index.js:172]: Yeah, I wasn't sure - felt weird there as a convention with all the imports. Should I make it all caps?

Comment (doom-weaver) [reply src/_http/index.js:172]: yeah that sounds like a good call.

Comment (doom-weaver) [reply src/_http/index.js:172]: you got another regex at line 147 btw.

Comment (doom-weaver) [reply src/_http/index.js:172]: actually reading a bit more: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions

it sounds like a literal will cause it to be compiled once.

So we may be good.

**Pull request closed without merging**